File: Task/ASCII-control-characters/Perl/ascii-control-characters.pl
use charnames ":loose";
# There is no EM, use END OF MEDIUM.
# Do not confuse BEL with BELL. Starting in Perl 5.18, BELL refers to unicode emoji 0x1F514. ALERT is an alias for BEL.
# compile time literal
"\N{nul}\N{soh}\N{stx}\N{etx}\N{eot}\N{enq}\N{ack}\N{bel}\N{bs}\N{ht}\N{lf}\N{vt}\N{ff}\N{cr}\N{so}\N{si}\N{dle}\N{dc1}\N{dc2}\N{dc3}\N{dc4}\N{nak}\N{syn}\N{etb}\N{can}\N{end of medium}\N{sub}\N{esc}\N{fs}\N{gs}\N{rs}\N{us}\N{space}\N{delete}"
# run time
charnames::string_vianame $_;

File: Task/ASCII-control-characters/Nim/ascii-control-characters.nim
import std/strutils

type AsciiControlChar {.pure.} = enum NUL, SOH, STX, ETX, EOT, ENQ, ACK, BEL,
                                      BS,  HT,  LF,  VT,  FF,  CR,  SO,  SI,
                                      DLE, DC1, DC2, DC3, DC4, NAK, SYN, ETB,
                                      CAN, EM,  SUB, ESC, FS,  GS,  RS,  US,
                                      SP,  DEL = 0x7F

echo ' ', CR, " = ", ord(CR).toHex(2)
echo DEL, " = ", ord(DEL).toHex(2)
